src/Controller/InAppPurchasesController.php

```diff
@@ -63,30 +69,45 @@ public function orderCart(RequestDataBag $data, Context $context): Response
     {
         $taxRate = \floatval($data->getString('taxRate'));
         $positions = $data->get('positions');
+        $extensionName = $data->get('name');
 
         $positionCollection = InAppPurchaseCartPositionCollection::fromArray($positions->all());
 
-        $validCartItems = $this->appPurchasesGateway->process(new InAppPurchasesPayload($positionCollection->getIdentifiers()));
+        $app = $this->getAppByName($extensionName, $context);
 
-        $filteredPositions = $positionCollection->filterValidInAppPurchases($positionCollection, $validCartItems->getPurchases());
+        if ($app) {
+            $payload = new InAppPurchasesPayload($positionCollection->getIdentifiers());
+            $validCartItems = $this->appPurchasesGateway->process($payload, $context, $app);
 
-        if ($filteredPositions->count() === 0) {
-            throw ExtensionStoreException::invalidInAppPurchase();
+            $positionCollection = $positionCollection->filterValidInAppPurchases($positionCollection, $validCartItems->getPurchases());
+
+            if ($positionCollection->count() === 0) {
+                throw ExtensionStoreException::invalidInAppPurchase();
+            }
         }
```

Comment:
I know this looks like duplicated code, but it's not. The collections are different. I can not place it behind an interface, because both classes might have the same method, but they dont have the same signatures. I can not place it behind an abstract class due to the same reason.